plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+const { merchant } = require( '@woocommerce/e2e-utils' );
+
+const { getRemotePluginZip, getLatestReleaseZipUrl, deleteDownloadedPluginFiles } = require( '@woocommerce/e2e-environment' );
+
+/**
+ * External dependencies
+ */
+const {
+	it,
+	beforeAll,
+} = require( '@jest/globals' );
+
+const { PLUGIN_REPOSITORY, PLUGIN_NAME, GITHUB_TOKEN } = process.env;
```

Comment:
Since this utility is Github specific `GITHUB_REPOSITORY` would be a better name than `PLUGIN_REPOSITORY`. Secondly, from a local command line perspective it would be nice to be able to add the repo and plugin name as parameters:

```
pnpx wc-e2e test:e2e tests/e2e/specs/smoke-tests/upload-plugin.js woocommerce/facebook-for-woocommerce 'Facebook for WooCommerce'
```

The parameters should be accessible through `process.env.argv`.

Reply:
Thanks for sharing this @rrennick! I pushed up the naming changes but when working with grabbing the parameters using `process.argv` they weren't being picked up for some reason:

```
    [
      '/.nvm/versions/node/v12.21.0/bin/node',
      '/woocommerce/node_modules/.pnpm/jest@25.5.4/node_modules/jest/bin/jest.js',
      '--maxWorkers=1',
      '--rootDir=./',
      '--verbose',
      'plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js',
      '--config=/woocommerce/plugins/woocommerce/tests/e2e/config/jest.config.js'
    ]
```

I'm just running this from `plugins/woocommerce` with the following:

```
pnpx wc-e2e test:e2e plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js woocommerce/facebook-for-woocommerce 'Facebook for WooCommerce'
```

I'll have to dig into this a bit to see why that may be.

Reply:
> We could also update the logic to allow all arguments (say, something like $@) but we'd need to change the logic for checking for a specific script:

@zhongruige This sounds good.

> There's also a few other tweaks since passing it in by name seems to break it apart:

Did the same happen with single quotes?

> I am thinking it may be worth breaking this out into a separate task

Sounds good.

Reply:
> Did the same happen with single quotes?

I did, I saw the same results using single and double quotes which was odd.

I created a new issue for this follow up work here: https://github.com/woocommerce/woocommerce/issues/31449